transformation-policy/src/test/java/io/apiman/plugins/transformation_policy/TransformationPolicyTest.java

```diff
@@ -0,0 +1,167 @@
+package io.apiman.plugins.transformation_policy;
+
+import static org.junit.Assert.*;
+import io.apiman.gateway.engine.beans.ServiceRequest;
+import io.apiman.gateway.engine.beans.ServiceResponse;
+import io.apiman.plugins.transformation_policy.transformer.XmlToJsonTransformer;
+import io.apiman.test.policies.ApimanPolicyTest;
+import io.apiman.test.policies.BackEndService;
+import io.apiman.test.policies.Configuration;
+import io.apiman.test.policies.IPolicyTestBackEndService;
+import io.apiman.test.policies.PolicyTestBackEndServiceResponse;
+import io.apiman.test.policies.PolicyTestRequest;
+import io.apiman.test.policies.PolicyTestRequestType;
+import io.apiman.test.policies.PolicyTestResponse;
+import io.apiman.test.policies.TestingPolicy;
+
+import java.io.UnsupportedEncodingException;
+
+import org.junit.Test;
+
+@SuppressWarnings("nls")
+@TestingPolicy(TransformationPolicy.class)
+public class TransformationPolicyTest extends ApimanPolicyTest {
+
+    @Test
+    @Configuration("{\"clientFormat\": \"XML\", \"serverFormat\": \"JSON\"}")
+    @BackEndService(ProduceJsonBackEndService.class)
+    public void transformServerJsonResponseToXml() throws Throwable {
+        PolicyTestRequest request = PolicyTestRequest.build(PolicyTestRequestType.GET, "/some/resource");
+        
+        PolicyTestResponse response = send(request);
+        
+        String expectedResponse = "<root><name>apiman</name></root>";
+        assertEquals("application/xml", response.header("Content-Type"));
+        assertNull(response.header("Content-Length"));
+        assertEquals(expectedResponse, response.body());
+    }
+
+    public static class ProduceJsonBackEndService implements IPolicyTestBackEndService {
```

Comment:
Can you break out the back-end service mock classes out into separate java files?  No big deal, just a personal preference. :)